sympy/physics/mechanics/kane.py

```diff
@@ -843,31 +891,40 @@ def kanes_equations(self, bodies, loads=None):
                     'kanes_equations(loads, bodies) > kanes_equations(bodies, loads).',
                     issue=10945, deprecated_since_version="1.1").warn()
 
+        # TODO : Why is it optional to pass in the kinematic differential
+        # equations? I'm not sure what the use case is without them.
         if not self._k_kqdot:
             raise AttributeError('Create an instance of KanesMethod with '
-                    'kinematic differential equations to use this method.')
+                                 'kinematic differential equations to use this '
+                                 'method.')
+
+        # TODO : Seems self._fr and self._frstar are set inside the following
+        # methods. Make it consistent with setting them in this method below.
         fr = self._form_fr(loads)
         frstar = self._form_frstar(bodies)
+
         if self._uaux:
-            if not self._udep:
+            if self.holonomic:
                 km = KanesMethod(self._inertial, self.q, self._uaux,
-                             u_auxiliary=self._uaux)
+                                 u_auxiliary=self._uaux)
             else:
                 km = KanesMethod(self._inertial, self.q, self._uaux,
-                        u_auxiliary=self._uaux, u_dependent=self._udep,
-                        velocity_constraints=(self._k_nh * self.u +
-                        self._f_nh))
+                                 u_auxiliary=self._uaux,
+                                 u_dependent=self._udep,
+                                 velocity_constraints=(self._k_nh * self.u +
+                                                       self._f_nh))
             km._qdot_u_map = self._qdot_u_map
             self._km = km
             fraux = km._form_fr(loads)
             frstaraux = km._form_frstar(bodies)
             self._aux_eq = fraux + frstaraux
             self._fr = fr.col_join(fraux)
             self._frstar = frstar.col_join(frstaraux)
+
         return (self._fr, self._frstar)
 
     def rhs(self, inv_method=None):
-        """ Returns the system's equations of motion in first order form.
+        """Returns the system's equations of motion in first order form.
```

Comment:
Returns the system's equations of motion in "explicit" first order form. For consistency in the labeling of forms done in the main docstring.